cmd/juju/controller/showcontroller.go

```diff
@@ -475,8 +475,14 @@ func (c *showControllerCommand) convertModelsForShow(
 	modelStatus []base.ModelStatus,
 ) {
 	controller.Models = make(map[string]ModelDetails)
+	if len(models) != len(modelStatus) {
+		controller.Errors = append(controller.Errors, "model status incomplete")
+	}
 	for i, m := range models {
 		modelDetails := ModelDetails{ModelUUID: m.UUID, OldModelUUID: m.UUID}
+		if i >= len(modelStatus) {
```

Comment:
Small nit on my part. Feel free to disregard.

Move this check to the top of the for loop. Alternatively make the models slice the same length as the modelStatus slice before the for loop and we can remove the defensive programming for each iteration.